### ISSUE TITLE
Java library for AWS Java SDK v2

### ISSUE THREAD

**Issue by jsyrjala** [NitorCreations/vault#41]
Implement java library for using AWS Java SDK v2
https://docs.aws.amazon.com/sdk-for-java/v2/developer-guide/welcome.html

Make separate maven library for this, e.g `com.nitorcreations:aws-vault-v2`  so that projects using v2 SDK do not have include v1 stuff.

**Comment by jsyrjala** [comment]
Let's also make VaultException to be RuntimeException.

**Comment by psiniemi** [comment]
This is now in master. Will release once we get node cli and integration tests that cover all the language versions via the cli

**Comment by jsyrjala** [comment]
I am wondering this difference:
- https://github.com/NitorCreations/vault/blob/master/java/src/main/java/com/nitorcreations/vault/VaultClient.java#L120
- https://github.com/NitorCreations/vault/blob/master/java-2/src/main/java/com/nitorcreations/vault2/VaultClient.java#L101

Default stack names are "vault" and "vault2". Why the difference? I think you should be able to use the same CloudFormation stack with both libs.

**Comment by psiniemi** [comment]
Yes, good catch. Refactoring bug... BTW this does not work yet - needs to be tested properly.

**Comment by jsyrjala** [comment]
Also old vault lib has a problem where VaultException is not thrown on all errors, some times you get AWS exceptions, like when key does not exist.
